File: plugins/dotnet/robocode.dotnet.core.net/src/Module.cs
using net.sf.robocode.serialization;
using robocode.dotnet.core.net.peer;

namespace robocode.dotnet.core.net
{
    public class Module
    {
        static Module()
        {
            RbSerializer.register(typeof (ExecCommands), RbSerializer.ExecCommands_TYPE);
            RbSerializer.register(typeof (BulletCommand), RbSerializer.BulletCommand_TYPE);
            RbSerializer.register(typeof (TeamMessage), RbSerializer.TeamMessage_TYPE);
            RbSerializer.register(typeof (DebugProperty), RbSerializer.DebugProperty_TYPE);
            RbSerializer.register(typeof (ExecResults), RbSerializer.ExecResults_TYPE);
            RbSerializer.register(typeof (BulletStatus), RbSerializer.BulletStatus_TYPE);
        }

        public static void Init()
        {
        }
    }
}

File: robocode.core/src/main/java/net/sf/robocode/core/Module.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package net.sf.robocode.core;


import net.sf.robocode.host.RobotStatics;
import net.sf.robocode.peer.*;
import net.sf.robocode.serialization.RbSerializer;
import net.sf.robocode.settings.ISettingsManager;
import net.sf.robocode.settings.SettingsManager;
import net.sf.robocode.version.IVersionManager;
import net.sf.robocode.version.VersionManager;

import java.util.List;


/**
 * @author Pavel Savara (original)
 */
public class Module extends BaseModule {
	static {
		Container.cache.addComponent(RobocodeMain.class);
		Container.cache.addComponent(IVersionManager.class, VersionManager.class);
		Container.cache.addComponent(ISettingsManager.class, SettingsManager.class);
	}

	@Override
	public void afterLoaded(List<IModule> allModules) {
		RbSerializer.register(ExecCommands.class, RbSerializer.ExecCommands_TYPE);
		RbSerializer.register(BulletCommand.class, RbSerializer.BulletCommand_TYPE);
		RbSerializer.register(TeamMessage.class, RbSerializer.TeamMessage_TYPE);
		RbSerializer.register(DebugProperty.class, RbSerializer.DebugProperty_TYPE);
		RbSerializer.register(ExecResults.class, RbSerializer.ExecResults_TYPE);
		RbSerializer.register(BulletStatus.class, RbSerializer.BulletStatus_TYPE);
		RbSerializer.register(RobotStatics.class, RbSerializer.RobotStatics_TYPE);
	}
}
